### DISCUSSION TITLE
My user_setup_select.h and user_setup.h seem to have no effect

### DISCUSSION THREAD

**Discussion by PeteHorn** [Bodmer/TFT_eSPI#2521] [Q&A - General] [answered]
Hi,

I'm 90% sure I'm missing something small (and probably obvious) here, but have been trying to get this to work for a couple of days and can't get anything except a blank white or black screen.

My hardware setup is an ESP32 dev board with a [ILI9488 display bought from ebay](https://www.ebay.co.uk/itm/174971594751).

My user_setup_select.h file has the root user_setup.h enabled. I've also tried enabling setup 21 (ESP32 + ILI9488) but that didn't make any difference.

![image](https://user-images.githubusercontent.com/9381579/226460609-89146f9c-9bf3-4625-89f6-ade36ad45963.png)

Then in user_setup.h I have the ILI9488 driver enabled
![image](https://user-images.githubusercontent.com/9381579/226461009-83bb94ae-63a9-45f3-9e51-30ab75de5ecb.png)

and the backlight configured to pin 27
![image](https://user-images.githubusercontent.com/9381579/226461090-f589ef53-0dc8-418f-b0df-9016618ae368.png)

And the SPI pins enabled for VSPI. 
![image](https://user-images.githubusercontent.com/9381579/226461150-c85b36bf-1106-4528-8291-6333b956b8e6.png)

I've checked all the hardware connections many times and its correct. I can enable pin 27 for the backlight in one of the example scripts to set it to high there, and the screen turns white instead of black - so I know the main script is getting to the ESP32.

However, when I run the read_user_setup example, I get this output

![image](https://user-images.githubusercontent.com/9381579/226461670-6bc91001-927a-48bc-80ef-7ba12301c8a0.png)

Display driver, display height, display width and TFT_DC are all wrong, so the settings I'm trying to apply aren't getting to the ESP32.

I found a few similar looking discussions, but everyone I saw had the correct output from read_user_setup, so I'm sure thats my first problem to sort out at least.

Any help would be much appreciated.

Thanks,

Pete

**Comment by Bodmer** [comment, marked as answer]
I am guessing you have not commented out this line:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L45

And have not commented out the lines here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L169

**Reply by PeteHorn** [reply]
Hi, thanks for the quick response, thats not it though.

![image](https://user-images.githubusercontent.com/9381579/226546901-22f56774-50a3-4d25-bdd0-10b26f4279b9.png)

![image](https://user-images.githubusercontent.com/9381579/226546961-d649fb13-a84f-41d4-b53b-25373698022b.png)

Is there any other potential causes you can think of?

Cheers, Pete

**Reply by Bodmer** [reply]
The User_Setup.h file must remain inside the library and the User_Setup_Select.h file must load it and no other setup file. At the moment the Read_User_Setup sketch is reporting the wrong driver and wrong pins, so the compiler is not finding a correct setup file.

**Reply by phorn-NI** [reply]
Thanks, that makes sense but what I can’t figure out is why it isn’t picking up the correct setup file when I haven’t moved any of the files on disk, and (I think) I’ve commented/uncommented lines of code correctly

**Reply by PeteHorn** [reply]
Figured it out and it was me missing something obvious. I have the repo cloned out to a separate folder and was updating the setup files there. Updating the files in the library folder worked straight away....

Thanks for the help, and for your effort to make the library
